### PR TITLE
SUBMARINE-580. Add an api for user login

### PR DESCRIPTION
**Pull request by Eroschang**
### What is this PR for?
Add an api for user login.

### What type of PR is it?
Improvement

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-580

### How should this be tested?
https://travis-ci.org/github/Eroschang/submarine/builds/716096128

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No

### CODE REVIEW

File: submarine-sdk/pysubmarine/setup.py [hunk 1]
```diff
@@ -40,6 +40,7 @@
         'python-dateutil >= 2.5.3',
         'pyarrow==0.17.0',
         'torch>=1.4.0',
+        'pyldap==2.4.37'
```
Comment (pingsutw) [inline submarine-sdk/pysubmarine/setup.py:43]:
```suggestion
       'python-ldap==3.3.1'
```

File: submarine-sdk/pysubmarine/github-actions/test-requirements.txt [hunk 2]
```diff
@@ -31,3 +31,4 @@ six >= 1.10
 python_dateutil >= 2.5.3
 setuptools >= 21.0.0
 urllib3 >= 1.15.1
+pyldap == 2.4.37
```
Comment (pingsutw) [inline submarine-sdk/pysubmarine/github-actions/test-requirements.txt:34]: Remove it

File: submarine-sdk/pysubmarine/submarine/experiment/api/experiment_client.py [hunk 3]
```diff
@@ -141,3 +142,17 @@ def list_log(self, status):
...
+    def login_ad(self, ldap_url, user_dn, password):
+        ld = ldap.initialize(ldap_url)
+        ld.protocol_version = 3
+        ld.set_option(ldap.OPT_REFERRALS, 0)
+        ld.simple_bind_s(user_dn, password)
+        cn = user_dn.split(',')[0].split('=')
+        base_dn = 'dc=testad,dc=com'
+        domain = 'testad.com'
+        username = '%s@%s' % (cn[1], domain)
+        ret = ld.search_s(base_dn, ldap.SCOPE_SUBTREE, "(userPrincipalName=%s" % username, ["userPrincipalName"])
```
Comment (pingsutw) [inline submarine-sdk/pysubmarine/submarine/experiment/api/experiment_client.py:155]:
```suggestion
        ret = ld.search_s(base_dn, ldap.SCOPE_SUBTREE,
                          "(userPrincipalName=%s" % username, ["userPrincipalName"])
```

**Comment by tangzhankun** [comment]
@Eroschang, this seems not a complete feature.

And a big problem is that checking user login on the client-side seems not secure. We should implement API in the submarine server and pass the user info to it from the client-side.

In my mind, there are three parts:
1. Some submarine server JAVA REST API:
Login/logout API
The login API should return a token so that the client can re-use it. Not pretty sure of this part.
2. submarine SDK:
generate the login/logout client Python library with the Swagger tool
3. Add login check in the experiment Python library
Add checking login logic before every experiment runs.

**Pull request closed without merging**